tests/ui/models/filter_alerts_with_videos_test.js

```diff
@@ -0,0 +1,103 @@
+import fetchMock from 'fetch-mock';
+
+import FilterAlertsWithVideos from '../../../ui/models/filterAlertsWithVideos';
+import testAlertSummaryWithVideos from '../mock/alerts_with_videos/alert_summary_with_browsertime_videos';
+import testAlertSummaryWithoutVideos from '../mock/alerts_with_videos/alert_summary_without_browsertime_videos';
+import currentJoblistWithVideoResultsOne from '../mock/alerts_with_videos/current_joblist_with_video_results_page_1';
+import currentJoblistWithVideoResultsTwo from '../mock/alerts_with_videos/current_joblist_with_video_results_page_2';
+import prevJoblistWithVideoResultsOne from '../mock/alerts_with_videos/prev_joblist_with_video_results_page_1';
+import prevJoblistWithVideoResultsTwo from '../mock/alerts_with_videos/prev_joblist_with_video_results_page_2';
+import currentJoblistWithoutVideoResultsOne from '../mock/alerts_with_videos/current_joblist_without_video_results_page_1';
+import currentJoblistWithoutVideoResultsTwo from '../mock/alerts_with_videos/current_joblist_without_video_results_page_2';
+import prevJoblistWithoutVideoResultsOne from '../mock/alerts_with_videos/prev_joblist_without_video_results_page_1';
+import prevJoblistWithoutVideoResultsTwo from '../mock/alerts_with_videos/prev_joblist_without_video_results_page_2';
+
+const dummyFrameworkName = 'browsertime';
+
+describe('FilterAlertsWithVideos', () => {
+  afterEach(() => {
+    fetchMock.reset();
+  });
+
+  describe('File bug url when alerts contain browsertime results links', () => {
+    beforeEach(() => {
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217`,
+        currentJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217&page=2`,
+        currentJoblistWithVideoResultsTwo,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998`,
+        prevJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998&page=2`,
+        prevJoblistWithVideoResultsTwo,
+      );
+    });
+
+    test('should return alerts with browsertime results links', async () => {
+      const alertsWithVideos = new FilterAlertsWithVideos(
+        testAlertSummaryWithVideos,
```

Comment:
Remember that we do in-place changes. We should provide a deep copy of the summary, otherwise this issue will surface when adding new tests.

Reply:
Can't picture how it will surface when adding new tests. `getFilteredAlerts()` will return an empty array if the framework isn't browsertime or there's no alert with video results within the items. `testAlertSummaryWithVideos` isn't used anywhere else in this test scenario.
Actually, there's an inconsistency in the logic. If we're doing in-place changes, doesn't really makes sense to return an empty array if no alerts have browsertime results. We could either return all the alerts and check that the code didn't enriched when no need or not do in-place changes at all. But if we return all the alerts, `getFilteredAlerts() name of the function isn't fit anymore. We could add another function `filterAndGetAllAlerts()` and make sure it will enrich the ones that have browsertime results only.